test/e2e/tests/test_open_id_connect_provider.py

```diff
@@ -0,0 +1,176 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	 http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""Integration tests for the IAM OIDCProvider resource"""
+
+import logging
+import time
+
+import pytest
+
+from acktest.k8s import condition
+from acktest.k8s import resource as k8s
+from acktest.resources import random_suffix_name
+from e2e import service_marker, CRD_GROUP, CRD_VERSION, load_resource
+from e2e.replacement_values import REPLACEMENT_VALUES
+from e2e import open_id_connect_provider
+from e2e import tag
+
+RESOURCE_PLURAL = "openidconnectproviders"
+
+DELETE_WAIT_AFTER_SECONDS = 5
+CHECK_STATUS_WAIT_SECONDS = 5
+MODIFY_WAIT_AFTER_SECONDS = 10
+
+
+@pytest.fixture
+def oidc_provider():
+    # required:
+    #   url
+    #   list of client IDs
+    #   list of server cert thumbprints (see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_providers_create_oidc_verify-thumbprint.html)
+    # optional:
+    #   a list of tags
+    oidc_provider_name = random_suffix_name("oidc-provider-ack-test", 24)
+
+    replacements = REPLACEMENT_VALUES.copy()
+    replacements["OPEN_ID_CONNECT_PROVIDER_NAME"] = oidc_provider_name
+
+    # the URL must begin with "https://"
+    # c.f. https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateOpenIDConnectProvider.html
+    replacements["URL"] = "https://host.domain.net"
```

Comment:
Please replace this URL with `example.com` - I can send you a guide on Amazon approved example strings

Reply:
Done